### DISCUSSION TITLE
How do I re-authenticate users?

### DISCUSSION THREAD

**Discussion by lexersodela** [pocketbase/pocketbase#2186] [Q&A] [answered]
I'm using the Dart SDK, I am allowing the **users(Auth Collection)** to use a collection by giving them token thru **auth with password** which gives me the token. Let's say that token expires, how do you re-authenticate the user? I tried using the **auth-refresh** but this only refresh authenticated users. I find it odd to save the user credential specially password in my shared preference then use it again if the tokens expires, I highly think that it's counter productive if my users need to log out first then log in again if their token expired, since my app do not involve financial or security risks.

Any work around for this?

**Comment by ganigeorgiev** [comment, marked as answer]
If the token has expired then you have to reauthenticate. This is not unique to PocketBase and you'll stumble on the same even if you were using an expired refresh token.

You can have a long lived auth token (the default duration can be changed from the Admin UI > Settings > Token options) and call `authRefresh()` on every app open.

Why PocketBase doesn't have a separate refresh and access tokens you can check in https://github.com/pocketbase/pocketbase/discussions/2154#discussioncomment-5426111.

**Reply by lexersodela** [reply]
> If the token has expired then you have to reauthenticate. This is not unique to PocketBase and you'll stumble on the same even if you were using an expired refresh token.

Do you have any suggestions in re-authentication? I can only think of log-out and log-in again which is kind of counter productive, I'm scoping this with 15-20 days expiration by the way, maybe I am overthinking it.

**Reply by ganigeorgiev** [reply]
> You can have a long lived auth token (the default duration can be changed from the Admin UI > Settings > Token options) and call authRefresh() on every app open.

**Comment by HEAVYPOLY** [comment]
@ganigeorgiev does this look correct for refreshing an auth token?
```
  if (pb.authStore.model) {
    user = pb.authStore.model
    await pb.collection('users').authRefresh()
  }
 ```

**Reply by ganigeorgiev** [reply]
A better approach would be to check whether the current store token is not expired, aka. using `pb.authStore.isValid`:

```js
if (pb.authStore.isValid) {
    await pb.collection('users').authRefresh();
}
```

**Comment by HEAVYPOLY** [comment]
Thanks, will use that instead.  And if not valid, show login ui if online
```javascript
    if (pb.authStore.isValid) {
      await pb.collection('users').authRefresh()
    } else if (window.navigator.onLine) {
      console.log("AUTHSTORE INVALID, SHOW LOGIN UI")
      showE('loginUI')
    }
```
